src/expr/src/scalar/func.rs

```diff
@@ -2040,34 +2040,16 @@ where
 }
 
 fn date_trunc_interval<'a>(a: Datum, int: Interval) -> Result<Datum<'a>, EvalError> {
```

Comment:
Looking at `date_trunc`, it splits validating the parameters and actually evaluating the function into two functions. We could follow the same pattern here with a `date_trunc_interval` plus an `_inner` function. On the other hand, it's pretty compact now without the match statement.

What about this to avoid `and_then`:
```
let unit = units.parse().map_err(...)?;
int.truncate_low_fields(unit, Some(0)).map_err(...)?
Ok(int.into())
```